### PR TITLE
Update the CI settings for R 4.3

### PR DESCRIPTION
**Pull request by yutannihilation**

### CODE REVIEW

**Comment by Ilia-Kosenkov** [comment]
are we dropping x86 completely?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -134,31 +133,9 @@ jobs:
...
           }
 
-          if ($env:RTOOLS_VERSION -ne '40') {
-            New-Item -Path libgcc_mock -Type Directory
-            New-Item -Path libgcc_mock\libgcc_eh.a -Type File
-            New-Item -Path libgcc_mock\libgcc_s.a -Type File
-
-            $pwd_slash = echo "${PWD}" | % {$_ -replace '\\','/'}
-            echo "LIBRARY_PATH=${pwd_slash}/libgcc_mock" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
-          }
```
Comment (Ilia-Kosenkov) [inline .github/workflows/test.yml:154]: This one is needed, but without `if` now

Comment (yutannihilation) [reply .github/workflows/test.yml:154]: Thanks for catching!

**Comment by yutannihilation** [comment]
> are we dropping x86 completely?

Yes! Why not?

**Comment by CGMossa** [comment]
We don't need libgcc mock anymore? What?

**Comment by yutannihilation** [comment]
> We don't need libgcc mock anymore? What?

Sorry, I was just confused. Restored it.

**Review by Ilia-Kosenkov — APPROVED** [review]

**Pull request merged**